factorial.sh

```
@@ -2,14 +2,22 @@
 
 factorial(){
     local num=$1
-    local a=1
-    while (($num > 1))
-        do
-        a=$(($a * $num))
-        ((num=$num-1))
-        done
-    echo $a
+    if (($num > 1)); then
+        local a=1
+        while (($num > 1))
+            do
+            a=$(($a * $num))
+            ((num=$num-1))
+            done
+    elif (($num = 1))
+    then
+        local a=1
+    elif (($num < 1))
+    then
+        a="the input must be higher than 1"
```

Good informative error message

---

How about raise error with exit code rather than printing